interfaces/builtin/zigbee_dongle.go

```diff
@@ -0,0 +1,137 @@
+}
+
+var zigbeeDongleDevPath = "/dev/zigbee/*"
+
+var zigbeeDonglePermanentSlotUdev = []byte(`
+IMPORT{builtin}="usb_id"
+SUBSYSTEM=="tty", SUBSYSTEMS=="usb", ATTRS{idProduct}=="0003", ATTRS{idVendor}=="10c4", SYMLINK+="zigbee/$env{ID_SERIAL}"
+`)
+
+// SanitizeSlot checks slot validity
+func (iface *ZigbeeDongleInterface) SanitizeSlot(slot *interfaces.Slot) error {
+	// check slot name
+	if iface.Name() != slot.Interface {
+		panic(fmt.Sprintf("slot is not of interface %q", iface))
+	}
+	return nil
+}
+
+// PermanentSlotSnippet - no permissions given to slot permanently
+func (iface *ZigbeeDongleInterface) PermanentSlotSnippet(slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor, interfaces.SecuritySecComp, interfaces.SecurityDBus:
+		return nil, nil
+	case interfaces.SecurityUDev:
+		return zigbeeDonglePermanentSlotUdev, nil
+	default:
+		return nil, interfaces.ErrUnknownSecurity
+	}
+}
+
+// ConnectedSlotSnippet - no permissions given to slot on connection
+func (iface *ZigbeeDongleInterface) ConnectedSlotSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor, interfaces.SecurityDBus, interfaces.SecuritySecComp, interfaces.SecurityUDev:
+		return nil, nil
+	default:
+		return nil, interfaces.ErrUnknownSecurity
+	}
+}
+
+// SanitizePlug checks plug validity
+func (iface *ZigbeeDongleInterface) SanitizePlug(slot *interfaces.Plug) error {
+	if iface.Name() != slot.Interface {
+		panic(fmt.Sprintf("plug is not of interface %q", iface))
+	}
+	return nil
+}
+
+// PermanentPlugSnippet no permissions provided to plug permanently
+func (iface *ZigbeeDongleInterface) PermanentPlugSnippet(plug *interfaces.Plug, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor, interfaces.SecuritySecComp, interfaces.SecurityDBus, interfaces.SecurityUDev:
+		return nil, nil
+	default:
+		return nil, interfaces.ErrUnknownSecurity
+	}
+}
+
+// ConnectedPlugSnippet returns security snippet specific to the plug
+func (iface *ZigbeeDongleInterface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		paths, err := iface.zigbeeDevPaths(slot)
+		if err != nil {
+			return nil, fmt.Errorf("cannot compute plug security snippet: %v", err)
+		}
+		var snippet bytes.Buffer
+		for _, path := range paths {
+			snippet.WriteString(fmt.Sprintf("%s rwk,\n", path))
+		}
+		return snippet.Bytes(), nil
+	case interfaces.SecuritySecComp, interfaces.SecurityDBus, interfaces.SecurityUDev:
+		return nil, nil
+	default:
+		return nil, interfaces.ErrUnknownSecurity
+	}
+}
+
+func (iface *ZigbeeDongleInterface) zigbeeDevPaths(slot *interfaces.Slot) ([]string, error) {
```

Comment:
As mentioned in private, my only issue with this pull request is that it grants access to all zigbee dongles. While I recognize the issue that currently we con't have a way to create dynamic slots (nothing like hot plug) I worry that later on we may have to say the `zigbee-dongle` interface is now constrained to a particular dongle (in the unlikely case of there being more than one). I'd prefer if we either 1) only allowed one (say, the one with the smallest serial number) or 2) renamed the interface to something broader like `zibgee-dongles`.